### PR TITLE
Fix/cr owner and others

### PR DESCRIPTION
**Pull request by jkmw**
## Common
- wait for harbor to become ready
- the parent instance will become the owner for all corresponding crs
- small refactoring of helper-types
- remove finalizers for replications and registries if they doesn't exist
- do not requeue a reconciled object if it has no errors

## Replication
- make IDs optional
- helm-chart fix
- wait for referenced registry to become ready

## User
- fix edge case of user creation: change password only if it has changed

### CODE REVIEW

File: pkg/controller/repository/repository_controller_test.go [hunk 1]
```diff
@@ -53,8 +53,8 @@ func TestRepositoryController_NonExistent_Instance(t *testing.T) {
 		t.Fatalf("reconcile returned error: (%v)", err)
 	}
 
-	if res.RequeueAfter != 30*time.Second {
-		t.Error("reconciliation did not requeue as expected")
+	if res.Requeue {
+		t.Error("reconciliation should not be re queued")
```
Comment (dkorittki) [inline pkg/controller/repository/repository_controller_test.go:57]:
```suggestion
		t.Error("reconciliation should not be requeued")
```

File: pkg/controller/repository/repository_controller_test.go [hunk 2]
```diff
@@ -77,7 +77,7 @@ func TestRepositoryController_Unready_Instance(t *testing.T) {
 	if err == nil {
 		t.Error("reconciliation did not return error as expected")
 	}
-	if !(res.RequeueAfter == 120*time.Second) {
+	if !(res.RequeueAfter == 30*time.Second) {
```
Comment (dkorittki) [inline pkg/controller/repository/repository_controller_test.go:80]:
```suggestion
	if !(res.RequeueAfter == 30 * time.Second) {
```

File: pkg/controller/user/user_controller_test.go [hunk 3]
```diff
@@ -55,7 +62,7 @@ func TestUserController_Empty_User_Spec(t *testing.T) {
 
 	res, err := r.Reconcile(req)
 	if err != nil {
-		t.Fatalf("reconcile returned error: (%v)", err)
+		t.Fatalf("reconcile returned no error: %s", err)
```
Comment (dkorittki) [inline pkg/controller/user/user_controller_test.go:65]: `%v` might show additional information, when `err` is a struct
```suggestion
		t.Fatalf("reconcile returned no error: %v", err)
```

File: pkg/controller/user/user_controller_test.go [hunk 4]
```diff
@@ -108,7 +115,7 @@ func TestUserController_Instance_Phase(t *testing.T) {
 		if err == nil {
 			t.Error("reconciliation did not return error as expected")
 		}
-		if res.RequeueAfter != 120*time.Second {
+		if res.RequeueAfter != 30*time.Second {
```
Comment (dkorittki) [inline pkg/controller/user/user_controller_test.go:118]:
```suggestion
		if res.RequeueAfter != 30 * time.Second {
```

File: pkg/controller/repository/repository_controller_test.go (hunk 1 above)
Comment (dkorittki) [reply pkg/controller/repository/repository_controller_test.go:57]: Nevermind, "re queued" also works.

File: pkg/controller/repository/repository_controller.go [hunk 5]
```diff
@@ -154,26 +157,29 @@ func (r *ReconcileRepository) Reconcile(request reconcile.Request) (reconcile.Re
...
 	}
 
-	return r.patchRepository(ctx, originalRepository, repository)
+	return r.updateRepositoryCR(ctx, harbor, originalRepository, repository, result)
 }
 
-// patchRepository compares the new CR status and finalizers with the pre-existing ones and updates them accordingly
-func (r *ReconcileRepository) patchRepository(ctx context.Context, originalRepository, repository *registriesv1alpha1.Repository) (reconcile.Result, error) {
+// updateRepositoryCR compares the new CR status and finalizers with the pre-existing ones and updates them accordingly
+func (r *ReconcileRepository) updateRepositoryCR(ctx context.Context, parentInstance *registriesv1alpha1.Instance, originalRepository, repository *registriesv1alpha1.Repository, result reconcile.Result) (reconcile.Result, error) {
```
[the remark this answers is not in the document]
Comment (jkmw) [reply pkg/controller/repository/repository_controller.go:183]: done

File: pkg/controller/replication/replication_controller.go [hunk 6]
```diff
@@ -151,25 +154,28 @@ func (r *ReconcileReplication) Reconcile(request reconcile.Request) (reconcile.R
...
+		result = reconcile.Result{}
 	}
-	return r.patchReplication(ctx, originalReplication, replication)
+	return r.updateReplicationCR(ctx, harbor, originalReplication, replication, result)
 }
 
-// patchReplication compares the new CR status and finalizers with the pre-existing ones and updates them accordingly
-func (r *ReconcileReplication) patchReplication(ctx context.Context, originalReplication, replication *registriesv1alpha1.Replication) (reconcile.Result, error) {
+// updateReplicationCR compares the new CR status and finalizers with the pre-existing ones and updates them accordingly
+func (r *ReconcileReplication) updateReplicationCR(ctx context.Context, parentInstance *registriesv1alpha1.Instance, originalReplication, replication *registriesv1alpha1.Replication, result reconcile.Result) (reconcile.Result, error) {
```
[the remark this answers is not in the document]
Comment (jkmw) [reply pkg/controller/replication/replication_controller.go:179]: done

File: pkg/controller/registry/registry_controller.go [hunk 7]
```diff
@@ -139,25 +144,28 @@ func (r *ReconcileRegistry) Reconcile(request reconcile.Request) (reconcile.Resu
...
 
 	}
-	return r.patchRegistry(ctx, originalRegistry, registry)
+	return r.updateRegistryCR(ctx, harbor, originalRegistry, registry, result)
 }
 
-// patchRegistry compares the new CR status and finalizers with the pre-existing ones and updates them accordingly
-func (r *ReconcileRegistry) patchRegistry(ctx context.Context, originalRegistry, registry *registriesv1alpha1.Registry) (reconcile.Result, error) {
+// updateRegistryCR compares the new CR status and finalizers with the pre-existing ones and updates them accordingly
+func (r *ReconcileRegistry) updateRegistryCR(ctx context.Context, parentInstance *registriesv1alpha1.Instance, originalRegistry, registry *registriesv1alpha1.Registry, result reconcile.Result) (reconcile.Result, error) {
```
[the remark this answers is not in the document]
Comment (jkmw) [reply pkg/controller/registry/registry_controller.go:169]: done

**Pull request merged**